pkg/kapis/alerting/v2alpha1/register.go

```diff
@@ -151,13 +165,27 @@ func AddToContainer(container *restful.Container, informers informers.InformerFa
 		Returns(http.StatusOK, ksapi.StatusOK, alertingv2alpha1.AlertList{}).
 		Metadata(restfulspec.KeyOpenAPITags, []string{constants.AlertingTag}))
 
+	ws.Route(ws.DELETE("/namespaces/{namespace}/rules").
+		To(handler.handleDeleteCustomAlertingRules).
+		Doc("delete multiple custom alerting rules in the specified namespace").
+		Param(ws.QueryParameter("name", "rule name").CollectionFormat(restful.CollectionFormatMulti).AllowMultiple(true)).
+		Returns(http.StatusOK, ksapi.StatusOK, alertingv2alpha1.BulkResponse{}).
+		Metadata(restfulspec.KeyOpenAPITags, []string{constants.AlertingTag}))
+
 	ws.Route(ws.POST("/namespaces/{namespace}/rules").
 		To(handler.handleCreateCustomAlertingRule).
 		Doc("create a custom alerting rule in the specified namespace").
 		Reads(alertingv2alpha1.PostableAlertingRule{}).
 		Returns(http.StatusOK, ksapi.StatusOK, "").
 		Metadata(restfulspec.KeyOpenAPITags, []string{constants.AlertingTag}))
 
+	ws.Route(ws.POST("/namespaces/{namespace}/bulk_rules").
+		To(handler.handleCreateOrUpdateCustomAlertingRules).
+		Doc("create or update custom alerting rules in bulk in the specified namespace").
+		Reads([]alertingv2alpha1.PostableAlertingRule{}).
+		Returns(http.StatusOK, ksapi.StatusOK, alertingv2alpha1.BulkResponse{}).
+		Metadata(restfulspec.KeyOpenAPITags, []string{constants.AlertingTag}))
+
 	ws.Route(ws.PUT("/namespaces/{namespace}/rules/{rule_name}").
```

Comment:
```suggestion
	ws.Route(ws.PUT("/namespaces/{namespace}/rules/{rule}").
```

Reply:
This will be held to distinguish another path parameter `rule_id` for builtin rules